pkg/instrumentation/auto/monitor.go

```diff
@@ -108,107 +151,212 @@ func NewMonitor(ctx context.Context, config MonitorConfig, k8sClient kubernetes.
 		logger.Error(err, "failed to start auto monitor")
 		return nil
 	}
-	factory.Start(ctx.Done())
-	synced := factory.WaitForCacheSync(ctx.Done())
-	for v, ok := range synced {
-		if !ok {
-			logger.Error(fmt.Errorf("caches failed to sync: %v", v), "bad cache sync")
+
+	// initialize workload factory before service factory so workloads are available during onServiceEvent calls when
+	// service informer is initialized
+	factories := []informers.SharedInformerFactory{workloadFactory, serviceFactory}
+
+	for _, factory := range factories {
+		factory.Start(ctx.Done())
+		synced := factory.WaitForCacheSync(ctx.Done())
+		for v, ok := range synced {
+			if !ok {
+				logger.Error(fmt.Errorf("workload caches failed to sync: %v", v), "bad cache sync")
```

Comment:
it's both the workload and service factories right? So not just workload

```suggestion
				logger.Error(fmt.Errorf("caches failed to sync: %v", v), "bad cache sync")
```